### PR TITLE
polish some well known unfriendly AAD server errors

### LINKED ISSUE #945: Need a better message to help diagnose the failure when a wrong user name is provided for az login
env) C:\azure-cli>az login -u <wrong username> -p <some>
WS-Trust RST request returned http error: 500 and server response: <s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://www.w3.org/2005/08/addressing" xmlns:u="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"><s:Header><a:Action s:mustUnderstand="1">http://www.w3.org/2005/08/addressing/soap/fault</a:Action><o:Security s:mustUnderstand="1" xmlns:o="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"><u:Timestamp u:Id="_0"><u:Created>2016-09-21T18:33:33.126Z</u:Created><u:Expires>2016-09-21T18:38:33.126Z</u:Expires></u:Timestamp></o:Security></s:Header><s:Body><s:Fault><s:Code><s:Value>s:Sender</s:Value><s:Subcode><s:Value xmlns:a="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd">a:FailedAuthentication</s:Value></s:Subcode></s:Code><s:Reason><s:Text xml:lang="en-US">ID3242: The security token could not be authenticated or authorized.</s:Text></s:Reason></s:Fault></s:Body></s:Envelope>

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #945

### CODE REVIEW

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py [hunk 1]
```diff
@@ -57,6 +57,13 @@ def login(username=None, password=None, service_principal=None, tenant=None):
             service_principal,
             tenant)
     except AdalError as err:
+        #try polish unfriendly server errors
+        msg = str(err)
+        suggestion = "For cross-check, try 'az login' to authenticate through browser"
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: What does 'cross-check' mean?

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py [hunk 2]
```diff
@@ -57,6 +57,13 @@ def login(username=None, password=None, service_principal=None, tenant=None):
...
+        if ('ID3242:' in msg) or ('Server returned an unknown AccountType' in msg):
+            raise CLIError("The user name might be invalid. " + suggestion)
+        if ('Server returned error in RSTR - ErrorCode' in msg) and username:
+            raise CLIError("Logining through command line is not supported. " + suggestion)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: Logining -> Logging in?

**Comment by derekbekoe** [comment]
Not sure why the CI build failed on Python 3.5-dev but if you restart that build, it should be fine.

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: Sounds good.

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: It means to trying login through browser which usually produces a better error.

**Pull request merged**